plugins/pulp_rpm/plugins/importers/yum/sync.py

```diff
@@ -1040,6 +1069,9 @@ def _identify_wanted_versions(self, package_info_generator):
             for unit, info in units.itervalues():
                 ret[unit] = info
 
+        if package_type == ids.TYPE_ID_RPM:
```

Comment:
I see the direction you're thinking. I'll try to put a different spin on it.

I think the best place to complain about data quality is as close as possible to where the data gets obtained, or used. If we re-wrote this, we'd probably want an object that owns the data, validation of it, and interaction with it. We'd put the validation there, and have that object do the validation at the right time.

Since we don't have that object, the next best thing (I think a least) is to put validation and use of the data close to each other.

Trying to do these behaviors at a higher level, I think might have the consequence of muddying the water in the middle. Given what we have, I think about it in roughly 3 layers. At the top, there's the ``run()`` method that orchestrates the high-level workflow. It calls into several middle-layer methods that orchestrate flow within a particular step. Still, these methods should be mostly about causing work to happen at the right time, and getting the right data to the right place for that to happen. At the bottom (leafs of the tree) is where the real work happens. If we can keep the top and middle focused on orchestration, that will enable people to most easily follow what's going on, and then re-assemble the small pieces later in a better way.

So that's all to say, I don't think we necessarily benefit from pushing "more important" actions higher up the code tree. Is that making any sense?